### DISCUSSION TITLE
Feature Request: ZEN mode for TOP

### DISCUSSION THREAD

**Discussion by Vincent0103** [TheOdinProject/theodinproject#4397] [Ideas]
### Checks

- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] The title of this issue follows the `Feature Request: brief description of feature request` format, e.g. `Feature Request: Add a dark mode to the website`
- [ ] Would you like to work on this issue?

### Description of the Feature Request

For peoples who are really inpatient while reading lessons, like I am. We usually tend to look at the lessons content to see what subsection we are currently, and it can be stressful, I found a way to sort of see only the content I want to see by zooming in by 175%. But it could be nice to have this feature to make it better for users to read lessons.
![image](https://github.com/TheOdinProject/theodinproject/assets/107498909/93570dd0-b03e-4c74-becf-dd4051647731)

The pictures are a way of the ZEN mode implementation on TOP.
![hashMapScreenshot](https://github.com/TheOdinProject/theodinproject/assets/107498909/1d258aac-05b9-4667-86fe-0c18dbb09969)
![hashMapScreenshot1](https://github.com/TheOdinProject/theodinproject/assets/107498909/f9437aa5-4658-4961-b514-40ec72602331)

### Acceptance criteria

- [ ] button to toggle between on/off for the ZEN mode.
- [ ] optionally, a shortcut to toggle ZEN mode.
- [ ] Ability to change the size of the text for readability. 

### Additional Comments

I love your work, without your free course I wouldn't know where I would have been. Thank you again for updating the course frequently, and have an amazing day!

**Comment by KevinMulhern** [comment]
Thanks for taking the time to leave this suggestion @Vincent0103. I like this idea and think it would be a good thing to have as a setting. 

But it may not be as easy to do as it looks. Because of the width constraints of our layout (the main body can never be wider than the navbar). The main content column on lesson pages is pulled to the left to leave space for the lesson navigation column.

Without the side navigation column to give balance, the page looks a bit lopsided aha:
<img width="1251" alt="Screenshot 2024-02-18 at 14 28 47" src="https://github.com/TheOdinProject/theodinproject/assets/7963776/38e87407-4b61-44cd-83a4-e9882dc4af0f">


It's especially noticeable at the bottom of lessons:
<img width="1861" alt="Screenshot 2024-02-18 at 14 44 56" src="https://github.com/TheOdinProject/theodinproject/assets/7963776/09b27880-2bc6-4aca-baac-c353538ea705">


We'd need to centre the main column if the sidebar isn't there. But knowing that part of the codebase, that might require a fairly big revision.

However, we do had plans for[ redoing the layout to open up the full width of the page](https://github.com/TheOdinProject/theodinproject/issues/3498). That feature is a priority for us this year and would make this feature request trivial to do.

In the meantime, I know its not ideal, but you can achieve something very similar to this by reducing the window size:
<img width="1405" alt="Screenshot 2024-02-18 at 14 50 37" src="https://github.com/TheOdinProject/theodinproject/assets/7963776/a4fa1699-0992-413c-94de-c1b1821dc3c9">

**Reply by Vincent0103** [reply]
Alright thank you for reviewing my request anyways! 😁

**Comment by Kezy-d4** [comment]
Hello there. Sorry to comment on an older issue, I have something to add and I don't think it warrants opening a separate issue.

In addition to the lesson contents being toggleable, I think special care should be given to ensure that the lesson contents are always available on any screen/window size, even if they are hidden by default on smaller screens. Currently, the lesson contents are entirely unavailable when working on devices with smaller screens or with a shrunken window. I personally rely on the lesson contents to help orient me during a lesson and having them completely unavailable when reading on my laptop or phone is a bit jarring. I've attached a short video demonstrating how MDN makes its article contents available on smaller screens, perhaps something similar could be implemented for TOP?
[mdn-sidebar.webm](https://github.com/user-attachments/assets/9b6d88ef-91b2-4955-b875-9d5e9ff4457f)
